### PR TITLE
Feature tanstack query

### PR DESCRIPTION
**Pull request by Julianna-Venter**
Added tanstack query of the user search page, there is still a lot of mock data for other features.

### CODE REVIEW

File: src/userForm/UserForm.tsx [hunk 1]
```diff
@@ -1,68 +1,143 @@
...
-                errors.username = "Invalid username address";
-              }
-              return errors;
-            }}
             onSubmit={(values, { setSubmitting }) => {
-              //when the form is submitted, the values are displayed in an alert
-              //write the values to json file if rememberMe is checked, and then that is the list of saved users that you will display on the saved menu
-              alert(JSON.stringify(values, null, 2));
+              console.log("Submitting form...");
+              console.log("Username:", selectedOption?.value);
```
Comment (tanadendere) [inline src/userForm/UserForm.tsx]: You forgot to remove your console.logs.

**Review by tanadendere — APPROVED** [review]
Looks good, I mostly focused on the UserForm page since that was what your PR was on. The tanstack query was added nicely. You just forgot some console.logs in your code but otherwise... well done!

File: src/userForm/UserForm.tsx [hunk 2]
```diff
@@ -1,68 +1,143 @@
...
+              console.log("Remember Me:", values?.rememberMe);
+              // Reset the form
+              navigate({ to: "/profile" });
+              //navigate({ to: '/posts/$postId', params: { postId } })
```
Comment (AyaazStreet) [inline src/userForm/UserForm.tsx:124]: Remember to remove commented out code (Or comment above if its for future use)

**Review by AyaazStreet — CHANGES_REQUESTED** [review]
Looks good just had one line of commented out code.

File: src/userForm/UserForm.tsx (hunk 2 above)
Comment (Julianna-Venter) [reply src/userForm/UserForm.tsx:124]: It's for future use, I've added the comment now

**Review by AyaazStreet — APPROVED** [review]

**Pull request merged**